packages/gateway/src/manager.ts

```diff
@@ -66,24 +66,44 @@ export function createGatewayManager(options: CreateGatewayManagerOptions): Gate
         )
       },
       async checkIfReshardingIsNeeded() {
-        gateway.logger.warn(`[Resharding] Checking if resharding is needed.`)
-        // Resharding is disabled.
-        if (!gateway.resharding.enabled) return { needed: false }
-        gateway.logger.warn(`[Resharding] Resharding is enabled.`)
+        if (!gateway.resharding.enabled) {
+          gateway.logger.debug('[Resharding] Resharding is disabled.')
+
+          return { needed: false }
+        }
+
+        gateway.logger.warn('[Resharding] Checking if resharding is needed.')
+        gateway.logger.warn('[Resharding] Resharding is enabled.')
 
         const sessionInfo = await gateway.resharding.getSessionInfo()
-        gateway.logger.warn(`[Resharding] Session info retrieved.`)
+
+        gateway.logger.warn('[Resharding] Session info retrieved.')
+        gateway.logger.debug(`[Resharding] Session info details: ${JSON.stringify(sessionInfo)}`)
+
         // Don't have enough identify limits to try resharding
-        if (sessionInfo.sessionStartLimit.remaining < sessionInfo.shards) return { needed: false, info: sessionInfo }
-        gateway.logger.warn(`[Resharding] Able to reshard, checking whether necessary now.`)
+        if (sessionInfo.sessionStartLimit.remaining < sessionInfo.shards) {
+          gateway.logger.warn('[Resharding] Not enough identify limits to try resharding.')
+
+          return { needed: false, info: sessionInfo }
+        }
+
+        gateway.logger.warn('[Resharding] Able to reshard, checking whether necessary now.')
 
         // 2500 is the max amount of guilds a single shard can handle
         // 1000 is the amount of guilds discord uses to determine how many shards to recommend.
         // This algo helps check if your bot has grown enough to reshard.
-        const percentage = ((2500 * sessionInfo.shards) / (gateway.totalShards * 1000)) * 100
+        // While this is imprecise as discord changes the recommended number of shard every 1000 guilds it is good enough
```

Comment:
Mention this in the comment of `shardsFullPercentage` so user can know that it is not 100% accurate

Reply:
https://github.com/discordeno/discordeno/pull/3792/commits/21592257219cd5e90b93c0dee8be6c2b52ca5444
maybe it is a bit too verbose? idk tell me if you think that it is